src/dlgConnectionProfiles.cpp

```diff
@@ -1787,37 +1764,35 @@ bool dlgConnectionProfiles::validateProfile()
             notificationAreaIconLabelError->show();
             notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(), tr("You have to enter a number. Other characters are not permitted.")));
             port_entry->setPalette(mErrorPalette);
-            validPort = false;
-            valid = false;
+            validProfile = false;
         }
 
         bool ok;
         int num = port.trimmed().toInt(&ok);
-        if (!port.isEmpty() && (num > 65536 && ok)) {
+        if (!port.isEmpty() && ((num >= 65536) || (num <= 0) || !ok)) {
```

Comment:
The check for negative numbers is *not* needed - as non digit characters including the minus sign will already have been filtered out by the test on old/new line 1783/1760 ... :nerd_face: 

... OTOH `0` is **not** a valid port number (it is reserved by IANA) so will need to be checked!